### PR TITLE
Model Settings Renderer Starting Point

### PR DESCRIPTION
**Pull request by rholinshead**
Model Settings Renderer Starting Point

# Model Settings Renderer Starting Point

Starting point for rendering prompt model settings. The SettingsRenderer is pretty much directly taken from lastmile without any significant changes; subsequent PRs will add additional renderers for types not yet supported (map, union)

## Testing:
- Supported settings render and correct values are persisted to the config:

https://github.com/lastmile-ai/aiconfig/assets/5060851/a35c43dc-9d0b-408e-9e4f-162c5e6a1342

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/545).
* __->__ #545
* #544

### CODE REVIEW

File: cli/aiconfig-editor/src/components/SettingsPropertyRenderer.tsx
Comment (saqadri) [inline cli/aiconfig-editor/src/components/SettingsPropertyRenderer.tsx:1]: This is the same from workbooks repo, right?

File: cli/aiconfig-editor/src/components/prompt/model_settings/ModelSettingsRenderer.tsx [hunk 1]
```diff
@@ -2,35 +2,32 @@ import ModelSettingsConfigRenderer from "@/src/components/prompt/model_settings/
...
   let settingsComponent;
 
   if (schema) {
     settingsComponent = (
-      <ModelSettingsSchemaRenderer settings={modelSettings} schema={schema} />
-    );
-  } else if (modelSettings) {
-    settingsComponent = (
-      <ModelSettingsConfigRenderer settings={modelSettings} />
+      <ModelSettingsSchemaRenderer
```
Comment (saqadri) [inline cli/aiconfig-editor/src/components/prompt/model_settings/ModelSettingsRenderer.tsx:23]: One thing that would use useful is to also add a JSON editor component as an option, to be able to explicitly edit the model settings JSON. Maybe just a Monaco editor? (Not as part of this diff)

**Review by saqadri — APPROVED** [review]
Looks good overall, just a few comments. One suggestion -- the width of the drawer is too narrow IMO, ideally it's a bit wider.

System prompt is also a small textbox -- we can iterate on this later!

File: cli/aiconfig-editor/src/components/prompt/model_settings/ModelSettingsRenderer.tsx (hunk 1 above)
Comment (rholinshead) [reply cli/aiconfig-editor/src/components/prompt/model_settings/ModelSettingsRenderer.tsx:23]: Yep, have that as a comment at the bottom of the component

Comment (rholinshead) [reply cli/aiconfig-editor/src/components/prompt/model_settings/ModelSettingsRenderer.tsx:23]: Leaving stuff like styling for later once we have functionality in place

**Pull request merged**